src/driver/amdxdna/aie2_pm.c

```diff
@@ -115,6 +115,31 @@ int aie2_pm_init(struct amdxdna_dev_hdl *ndev)
 	return 0;
 }
 
+int npu1_get_tops(struct amdxdna_dev_hdl *ndev, u64 *max, u64 *curr)
+{
+	u64 total_col, hclk_freq;
+
+	total_col = ndev->total_col;
+	hclk_freq = ndev->hclk_freq;
+	*max = 2 * ndev->total_col;
+	*curr = hclk_freq * 2 * total_col / 1028;
+
+	return 0;
+}
+
+int npu4_get_tops(struct amdxdna_dev_hdl *ndev, u64 *max, u64 *curr)
+{
+	const struct amdxdna_dev_priv *pdev = ndev->priv;
+	u64 total_col, hclk_freq;
+
+	total_col = ndev->total_col;
+	hclk_freq = ndev->hclk_freq;
+	*max = 4096 * total_col * pdev->dpm_clk_tbl[ndev->max_dpm_level].hclk / 1000000;
+	*curr = 4096 * hclk_freq * total_col / 1000000;
```

Comment:
Looks like both formula has same component. Can you try?
```suggestion
        topc = 4096 * total_col / 1000000;
	*max = topc * pdev->dpm_clk_tbl[ndev->max_dpm_level].hclk;
	*curr = topc * hclk_freq;
```

Reply:
I'm also thinking, the "topc" and "max tops" are constant value for a device instance. Can we calculate it once in the aie2_pm_init(). Is it better?